packages/flutter_tools/lib/src/ios/migrations/project_build_location_migration.dart

```diff
@@ -0,0 +1,49 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import '../../base/file_system.dart';
+import '../../base/logger.dart';
+import '../../project.dart';
+import 'ios_migrator.dart';
+
+// Update the xcodeproj build location. Legacy build location does not work with Swift Packages.
+class ProjectBuildLocationMigration extends IOSMigrator {
+  ProjectBuildLocationMigration(
+    IosProject project,
+    Logger logger,
+  ) : _xcodeProjectWorkspaceData = project.xcodeProjectWorkspaceData,
+      super(logger);
+
+  final File _xcodeProjectWorkspaceData;
+
+  @override
+  bool migrate() {
+    if (!_xcodeProjectWorkspaceData.existsSync()) {
+      logger.printTrace('Xcode project workspace data not found, skipping build location migration.');
+      return true;
+    }
+
+    processFileLines(_xcodeProjectWorkspaceData);
+    return true;
+  }
+
+  @override
+  String migrateLine(String line) {
+    const String legacyBuildLocation = 'location = "group:Runner.xcodeproj"';
+    const String defaultBuildLocation = 'location = "self:"';
+
+    return line.replaceAll(legacyBuildLocation, defaultBuildLocation);
+  }
+
+  @override
+  String migrateFileContents(String fileContents) {
+    const String podLocation = '''
```

Comment:
nit. is this guaranteed to match, even with the surrounding whitespace?

Reply:
We generated this as part of the template with https://github.com/flutter/flutter/pull/4893/files#diff-fddbdff9cdb9e32ee9b9a95f74f30788R5-R8.  Xcode and CocoaPods won't add it (at least how we have the project set up now).